### PR TITLE
SW-5795 Use dynamic IDs in CSV importer tests

### PR DESCRIPTION
**Pull request by sgrimm**
As a step toward supporting parallel test runs, update `BatchImporterTest` and
`SpeciesImporterTest` to stop using hardwired IDs. These tests make use of the
new `mapTo1IndexedIds` helper function to verify that the imported database rows
refer to each other correctly.

### CODE REVIEW

**Comment by sgrimm** [comment]
* **#2271** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2271?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2270** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2270?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2269** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2269?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2268** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2268?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2267** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2267?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2266** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2266?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sgrimm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by nickgraz — APPROVED** [review]
Looks good to me

File: src/test/kotlin/com/terraformation/backend/species/db/SpeciesImporterTest.kt [hunk 1]
```diff
@@ -659,29 +650,38 @@ internal class SpeciesImporterTest : DatabaseTest(), RunsAsUser {
...
+            failingSpeciesStore,
+            uploadProblemsDao,
+            uploadsDao,
+            uploadService,
+            uploadStore,
+            userStore)
+    every { failingSpeciesStore.importSpecies(any(), any()) } throws RuntimeException("Failed!")
     every { fileStore.read(storageUrl) } returns
-        sizedInputStream("$header\nNew name,Common,Family,CR,false,Shrub,Recalcitrant,")
+        sizedInputStream("$header\nNew name,Common,Family,CR,false,Shrub,Recalcitrant,Tundra")
```
Comment (nickgraz) [inline src/test/kotlin/com/terraformation/backend/species/db/SpeciesImporterTest.kt:675]: This was just to make it closer to our real data, right? Or was it failing to fail without that?

Comment (sgrimm) [reply src/test/kotlin/com/terraformation/backend/species/db/SpeciesImporterTest.kt:675]: Yeah, just to make it closer to real.

**Comment by sgrimm** [comment]
### Merge activity

* **Aug 12, 4:51 PM EDT**: @sgrimm started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2270).
* **Aug 12, 5:29 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2270) rebased this pull request as part of a merge.
* **Aug 12, 5:38 PM EDT**: @sgrimm merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2270).

**Pull request merged**